pyquil/api/_qpu.py

```diff
@@ -288,22 +218,21 @@ def _build_patch_values(self) -> Dict[str, List[Union[int, float]]]:
         self._update_variables_shim_with_recalculation_table()
 
         # Initialize our patch table
-        recalculation_table = getattr(self._executable, "recalculation_table", None)
-        if recalculation_table is not None:
-            memory_ref_names = list(set(mr.name for mr in recalculation_table.keys()))
-            if memory_ref_names != []:
-                assert len(memory_ref_names) == 1, (
-                    "We expected only one declared memory region for "
-                    "the gate parameter arithmetic replacement references."
-                )
-                memory_reference_name = memory_ref_names[0]
-                patch_values[memory_reference_name] = [0.0] * len(recalculation_table)
-
-        assert isinstance(self._executable, QuiltBinaryExecutableResponse)
-        for name, spec in self._executable.memory_descriptors.items():
+        assert isinstance(self.executable, EncryptedBinary)
+        recalculation_table = self.executable.recalculation_table
+        memory_ref_names = list(set(mr.name for mr in recalculation_table.keys()))
+        if memory_ref_names:
+            assert len(memory_ref_names) == 1, (
+                "We expected only one declared memory region for "
+                "the gate parameter arithmetic replacement references."
+            )
+            memory_reference_name = memory_ref_names[0]
+            patch_values[memory_reference_name] = [0.0] * len(recalculation_table)
+
+        for name, spec in self.executable.memory_descriptors.items():
             # NOTE: right now we fake reading out measurement values into classical memory
             # hence we omit them here from the patch table.
-            if any(name == mref.name for mref, _ in self._executable.ro_sources):
+            if any(name == mref.name for mref, _ in self.executable.ro_sources.items()):
```

Comment:
nit - don't need `items()` if you're ignoring the value

Reply:
Done.